og_access/og_access.api.php

```diff
@@ -14,22 +14,51 @@
 /**
  * Allow modules to mark group privacy change.
  *
- * @param &$context
- *   Array including 3 keys:
- *   'entity' - the submitted group entity.
- *   'entity_type' - String. The type of the entity that was submitted.
- *   'change_detected' - Boolean. Marks whether the group privacy changed or not.
- *                       FALSE by default, and changeable be each hook
- *                       implementation.
+ * @param array $context
+ *   Array with the following keys:
+ *   'entity': the group entity.
+ *   'entity_type':  The group type.
+ *
+ * @return bool
+ *   TRUE if group privacy has been changed.
  */
-function hook_og_access_invoke_node_access_acquire_grants_alter(&$context) {
+function hook_og_access_invoke_node_access_acquire_grants($context) {
   $wrapper = entity_metadata_wrapper($context['entity_type'], $context['entity']);
+  if (!isset($wrapper->OG_ACCESS_FIELD)) {
+    // Group doesn't have OG access field attached to it.
+    return;
+  }
+
+  $original_wrapper = entity_metadata_wrapper($context['entity_type'], $context['entity']->original);
+
+  $og_access = $wrapper->{OG_ACCESS_FIELD}->value();
+  $original_og_access = $original_wrapper->{OG_ACCESS_FIELD}->value();
+  return $og_access !== $original_og_access;
+}
+
+/**
+ * Allow modules to mark group privacy change.
+ *
+ * @param $result
+ *   Array of results of the hooks called before.
+ * @param array $context
+ *   Array with the following keys:
+ *   'entity': the group entity.
+ *   'entity_type':  The group type.
+ */
+function hook_og_access_invoke_node_access_acquire_grants_alter(&$result, $context) {
+  $wrapper = entity_metadata_wrapper($context['entity_type'], $context['entity']);
```

Comment:
I think example should be:
```php
if ($access !== $original_access) {
  $result['my_module'] = TRUE; 
}
```
Because after altering all functions we are checking that we have in results at least one ``TRUE``, that means that access has changed and we go on.